Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.m

```diff
@@ -359,72 +353,124 @@ - (void)configureViews
+        [textViews addObject:self.footerView];
     }
 
-    if (textStackView) {
-        CGSize bubbleSize = [self measureSize];
-        [NSLayoutConstraint autoSetPriority:UILayoutPriorityRequired
-                             forConstraints:^{
-                                 [self.viewConstraints addObjectsFromArray:@[
-                                     [self autoSetDimension:ALDimensionWidth toSize:bubbleSize.width],
-                                 ]];
-                             }];
+    [self insertTextViewsIntoStackViewIfNecessary:textViews];
+
+    CGSize bubbleSize = [self measureSize];
+    [NSLayoutConstraint autoSetPriority:UILayoutPriorityRequired
+                         forConstraints:^{
+                             [self.viewConstraints addObjectsFromArray:@[
+                                 [self autoSetDimension:ALDimensionWidth toSize:bubbleSize.width],
+                             ]];
+                         }];
+
+    [self updateBubbleColorWithBodyMediaView:bodyMediaView];
+}
+
+- (void)updateBubbleColorWithBodyMediaView:(nullable UIView *)bodyMediaView
+{
+    BOOL hasOnlyBodyMediaView = NO;
+    switch (self.cellType) {
+        case OWSMessageCellType_Unknown:
+        case OWSMessageCellType_TextMessage:
+        case OWSMessageCellType_OversizeTextMessage:
+        case OWSMessageCellType_GenericAttachment:
+        case OWSMessageCellType_DownloadingAttachment:
+        case OWSMessageCellType_ContactShare:
+            break;
+        case OWSMessageCellType_StillImage:
+        case OWSMessageCellType_AnimatedImage:
+        case OWSMessageCellType_Audio:
+        case OWSMessageCellType_Video:
+            hasOnlyBodyMediaView = (bodyMediaView && self.stackView.subviews.count == 1);
+            break;
+    }
+    if ([self.viewItem.interaction isKindOfClass:[TSMessage class]] && !hasOnlyBodyMediaView) {
+        TSMessage *message = (TSMessage *)self.viewItem.interaction;
+        self.bubbleView.bubbleColor = [self.bubbleFactory bubbleColorWithMessage:message];
+    } else {
+        // Media-only messages should have no background color; they will fill the bubble's bounds
+        // and we don't want artifacts at the edges.
+        self.bubbleView.bubbleColor = nil;
     }
 }
 
+- (BOOL)canFooterOverlayMedia
+{
+    switch (self.cellType) {
+        case OWSMessageCellType_Unknown:
+        case OWSMessageCellType_TextMessage:
+        case OWSMessageCellType_OversizeTextMessage:
+            return NO;
+        case OWSMessageCellType_StillImage:
+        case OWSMessageCellType_AnimatedImage:
+        case OWSMessageCellType_Video:
+            return YES;
+        case OWSMessageCellType_Audio:
+        case OWSMessageCellType_GenericAttachment:
+        case OWSMessageCellType_DownloadingAttachment:
+        case OWSMessageCellType_ContactShare:
+            return NO;
+    }
+}
+
+- (void)insertTextViewsIntoStackViewIfNecessary:(NSArray<UIView *> *)textViews
```

Comment:
TIOLI naming nit: `insertAnyTextViewsIntoStackView:` to succinctly surface what is "necessary" for them to be added.

Reply:
Done.